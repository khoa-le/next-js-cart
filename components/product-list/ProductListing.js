import styled from "styled-components";
import { GlobalConfiguration } from "../../utils/global";
import ProductListingItem from "./ProductListingItem";
import Pagination from "./Pagination";

const ProductListingWrapper = styled.div``;

const ProductListing = ({ products }) => {
  return (
    <div className="container">
      <ProductListingWrapper className="row mx-gutters-5">
        {products.length == 0 ? (
          <p>Product listing was empty</p>
        ) : (
          <>
            {Object.keys(products).map((key) => {
              if (key < GlobalConfiguration.ProductListing.PerPage) {
                const product = products[key];
                return (
                  <ProductListingItem
                    product={product}
                    key={product.product_id}
                  />
                );
              }
            })}
          </>
        )}
      </ProductListingWrapper>
      <Pagination numberProducts={products.length}></Pagination>
    </div>
  );
};
export default ProductListing;
