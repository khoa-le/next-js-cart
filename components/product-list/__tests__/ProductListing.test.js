import { render, cleanup } from "@testing-library/react";
import ProductListing from "../ProductListing";
import * as nextRouter from "next/router";

nextRouter.useRouter = jest.fn();
nextRouter.useRouter.mockImplementation(() => ({
   query: { page: 1 } ,
   asPath: "/products"
  }));

afterEach(cleanup);

const products = [
  {
    product_id: 1,
    image: "12312321",
    name: "Khoa Product",
    price: 4321,
    discounted_price: 0,
  },
  {
    product_id: 2,
    image: "12312123321",
    name: "Khoa Product 1",
    price: 11,
    discounted_price: 0,
  },
];

describe("ProductListing", () => {
  test("Can render Product List 123", () => {
    const { container, getByText } = render(
      <ProductListing products={products} />
    );
    const productName1Element = getByText("Khoa Product");
    const productName2Element = getByText("Khoa Product 1");

    expect(container.firstChild).toMatchSnapshot();
    expect(productName1Element).toBeInTheDocument();
    expect(productName2Element).toBeInTheDocument();
  });
  test("Product List Empty", () => {
    const { container, getByText } = render(<ProductListing products={[]} />);
    const messageEmptyElement = getByText("Product listing was empty");
    expect(messageEmptyElement).toBeInTheDocument();
  });
});
