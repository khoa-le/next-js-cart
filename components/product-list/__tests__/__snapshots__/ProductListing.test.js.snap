// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListing Can render Product List 123 1`] = `
<div
  class="container"
>
  <div
    class="sc-AxhCb ccMvAG row mx-gutters-5"
  >
    <div
      class="sc-AxjAm gxygnu col-sm-6 col-lg-4 mb-9"
    >
      <a
        class="d-block text-dark text-center transition-3d-hover"
        href="/products/1"
      >
        <img
          alt="Khoa Product"
          class="img-fluid rounded"
          src="/media/product_images/12312321"
        />
        <span
          class="pt-4 px-4"
        >
          <h2
            class="h5 mb-1"
          >
            Khoa Product
          </h2>
          <span
            class="d-block"
          >
            4321
          </span>
        </span>
      </a>
    </div>
    <div
      class="sc-AxjAm gxygnu col-sm-6 col-lg-4 mb-9"
    >
      <a
        class="d-block text-dark text-center transition-3d-hover"
        href="/products/2"
      >
        <img
          alt="Khoa Product 1"
          class="img-fluid rounded"
          src="/media/product_images/12312123321"
        />
        <span
          class="pt-4 px-4"
        >
          <h2
            class="h5 mb-1"
          >
            Khoa Product 1
          </h2>
          <span
            class="d-block"
          >
            11
          </span>
        </span>
      </a>
    </div>
  </div>
  <div
    class="sc-AxiKw cxroOZ"
  >
    <nav
      aria-label="Page navigation"
    >
      <ul
        class="pagination mb-0"
      />
    </nav>
  </div>
</div>
`;
